### PR TITLE
[BE] [FIX] use correct dish object to remapp allergens

### PR DESCRIPTION
**Pull request by MarcPIster**
### 🛠 Changes being made

ensure that we take the correct id of the dish.
add combo to ensure that we dont delete it

### ✨ What's the context?

we want to be able to remap dishes if allergens change 

### 🧠 Rationale behind the change

need to show all allergens correctly
### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/backend/src/controllers/dishesController.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ export async function changeDishByID(
   const newDish: IDishBE = {
     //if the new dish has a property, use it, else use the old one
     name: dish.name ? dish.name : oldDish.name as string,
-    uid: oldDish.uid as number,
+    uid: dish.uid ? dish.uid : oldDish.uid as number,
```
[the remark this answers is not in the document]
Comment (MarcPIster) [reply packages/backend/src/controllers/dishesController.ts:324]: no, because this solve the bug. sometime oldDish dosnt has a uid.. thats why i take the new one

**Review by JosefineM2000 — APPROVED** [review]
